tests/foreman/sys/test_fam.py

```diff
@@ -178,6 +187,72 @@ def create_fake_module(module_target_sat, module_name, module_classes):
     create_fake_module(module_target_sat, 'fakemodule', ['init'])
 
 
+@pytest.fixture(scope='module')
+def setup_fam_with_idm(idm_sat, module_sca_manifest):
+    # Execute AAP WF for FAM setup
+    Broker().execute(workflow='fam-test-setup', source_vm=idm_sat.name)
+
+    # Modify and copy config files to the Satellite
+    idm_fam_settings = settings.fam.server.copy()
+    # Replace hostname in the settings
+    for k, v in idm_fam_settings.items():
+        if isinstance(v, str):
+            idm_fam_settings[k] = v.replace(settings.server.hostname, idm_sat.hostname)
+    # Remove username and password as they are not used in IDM
+    idm_fam_settings['foreman_username'] = "{{ omit }}"
+    idm_fam_settings['satellite_username'] = "{{ omit }}"
+    idm_fam_settings['foreman_password'] = "{{ omit }}"
+    idm_fam_settings['satellite_password'] = "{{ omit }}"
+
+    idm_sat.put(
+        idm_fam_settings.to_yaml(),
+        f'{FAM_ROOT_DIR}/tests/test_playbooks/vars/server.yml',
+        temp_file=True,
+    )
+
+    common_fam_setup(idm_sat)
+
+    # Upload manifest to test playbooks directory
+    idm_sat.put(str(module_sca_manifest.path), str(module_sca_manifest.name))
+    idm_sat.execute(f'mv {module_sca_manifest.name} {FAM_ROOT_DIR}/tests/test_playbooks/data')
+    config_file = f'{FAM_ROOT_DIR}/tests/test_playbooks/vars/server.yml'
+    idm_sat.execute(
+        f'''sed -i 's|subscription_manifest_path:.*|subscription_manifest_path: "data/{module_sca_manifest.name}"|g' {config_file}'''
+    )
+
+
+@pytest.fixture(scope='module')
+def idm_sat(satellite_factory, ad_data):
```

Comment:
We already have `module_enroll_idm_and_configure_external_auth`. You stated you have some reason not to use it, can you just write it here?

Reply:
Well, I made the decision some time ago and can't remember the main reason...
I use own satellite_factory for some reason instead of module_target_sat (which is used by the mentioned fixture). It could be probably re-written, but it would require more effort.
But thanks to your comment I discovered that I forgot to call `disenroll_idm`, so I fixed it.

Reply:
You have to use factory otherwise you would modify module_target_sat permanently to use IdM always the you couldn't test non-IdM setup  (user playbooks)